File: backend/src/main/java/ropold/backend/dto/AssignmentDTO.java
package ropold.backend.dto;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public record AssignmentDTO(
        UUID id,
        DeviceDTO device,
        EmployeeDTO employee,
        EmployeeDTO handedOutBy,
        LocalDate assignedDate,
        LocalDate returnedDate,
        String conditionOut,
        String conditionIn,
        String notes,
        boolean copyHandedToEmployee,
        boolean copyFiledInPersonnelFile,
        List<AssignmentFileDTO> files
) {}

File: frontend/src/components/dto/AssignmentDTO.ts
import { type DeviceDTO } from "./DeviceDTO.ts";
import { type EmployeeDTO } from "./EmployeeDTO.ts";
import { type AssignmentFileDTO } from "./AssignmentFileDTO.ts";

export type AssignmentDTO = {
    id: string | null;
    device: DeviceDTO;
    employee: EmployeeDTO;
    handedOutBy: EmployeeDTO | null;
    assignedDate: string;
    returnedDate: string | null;
    conditionOut: string | null;
    conditionIn: string | null;
    notes: string | null;
    copyHandedToEmployee: boolean;
    copyFiledInPersonnelFile: boolean;
    files: AssignmentFileDTO[] | null;
};